From 990b368672cccb1c1c9fe189517abedf5455cb5c Mon Sep 17 00:00:00 2001
From: Eduardo Habkost <ehabkost@redhat.com>
Date: Tue, 25 Mar 2014 16:03:46 +0100
Subject: [PATCH 1/2] pc: Use cpu64-rhel6 CPU model by default on rhel6 machine-types

RH-Author: Eduardo Habkost <ehabkost@redhat.com>
Message-id: <1395763427-31684-2-git-send-email-ehabkost@redhat.com>
Patchwork-id: 58246
O-Subject: [RHEL7 qemu-kvm PATCH 1/2] pc: Use cpu64-rhel6 CPU model by default on rhel6 machine-types
Bugzilla: 1080170
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>
RH-Acked-by: Alex Williamson <alex.williamson@redhat.com>
RH-Acked-by: Bandan Das <bsd@redhat.com>

Bugzilla: 1080170
Brew scratch build: http://brewweb.devel.redhat.com/brew/taskinfo?taskID=7250917
Upstream status: not applicable

On RHEL-6, cpu64-rhel6 was the default CPU model, so use it as default
on rhel6.* machine-types so we keep the ABI and not break RHEL6->RHEL7
live-migration.

Signed-off-by: Eduardo Habkost <ehabkost@redhat.com>
---
 hw/i386/pc_piix.c | 3 +++
 1 file changed, 3 insertions(+)

Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 hw/i386/pc_piix.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/hw/i386/pc_piix.c b/hw/i386/pc_piix.c
index be68098..940816f 100644
--- a/hw/i386/pc_piix.c
+++ b/hw/i386/pc_piix.c
@@ -917,6 +917,9 @@ static QEMUMachine pc_machine_rhel700 = {
 static void pc_compat_rhel650(QEMUMachineInitArgs *args)
 {
     pc_compat_rhel700(args);
+    if (!args->cpu_model) {
+        args->cpu_model = "cpu64-rhel6";
+    }
     x86_cpu_compat_set_features("pentium", FEAT_1_EDX, 0, CPUID_APIC);
     x86_cpu_compat_set_features("pentium2", FEAT_1_EDX, 0, CPUID_APIC);
     x86_cpu_compat_set_features("pentium3", FEAT_1_EDX, 0, CPUID_APIC);
-- 
1.7.1

